src/lists/list.jsx

```diff
@@ -92,35 +95,32 @@ const List = React.createClass({
       ...other,
     } = this.props;
 
+    let hasSubheader = false;
+
+    if (subheader) {
+      hasSubheader = true;
+    } else {
+      const firstChild = React.Children.toArray(children)[0];
+      if (React.isValidElement(firstChild) && firstChild.type === Subheader) {
```

Comment:
Actually, I see one drawback here. We end up importing the `<Subheader />` component even if we don't need it.
What about using `displayName` :grin:?